pkg/cache/context_test.go

```diff
@@ -460,3 +464,60 @@ func TestRemoveTask(t *testing.T) {
 	// now there is no task left
 	assert.Equal(t, len(app.GetNewTasks()), 0)
 }
+
+func TestPublishEvents(t *testing.T) {
```

Comment:
we should have another UT to cover the case where app or task doesn't exist in the context.

Reply:
right. added a negative test